src/Compilers/VisualBasic/Portable/Emit/SymbolTranslator.vb

```diff
@@ -203,21 +203,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
             Return namedTypeSymbol
         End Function
 
-        Private Sub CheckTupleUnderlying(namedTypeSymbol As NamedTypeSymbol, syntaxNodeOpt As SyntaxNode, diagnostics As DiagnosticBag)
+        Private Sub CheckTupleUnderlyingType(namedTypeSymbol As NamedTypeSymbol, syntaxNodeOpt As SyntaxNode, diagnostics As DiagnosticBag)
             ' check that underlying type of a ValueTuple is indeed a value type (or error)
             ' this should never happen, in theory,
             ' but if it does happen we should make it a failure.
             ' NOTE: declaredBase could be null for interfaces
             Dim declaredBase = namedTypeSymbol.BaseTypeNoUseSiteDiagnostics
-            If declaredBase Is Nothing OrElse declaredBase.SpecialType <> SpecialType.System_ValueType Then
-                ' Try to decrease noise by not complaining about the same type over and over again.
-                If (_reportedErrorTypesMap.Add(namedTypeSymbol)) Then
-                    diagnostics.Add(New VBDiagnostic(
-                                    ErrorFactory.ErrorInfo(ERRID.ERR_PredefinedValueTupleTypeMustBeStruct, namedTypeSymbol.MetadataName),
-                                    If(syntaxNodeOpt Is Nothing, NoLocation.Singleton, syntaxNodeOpt.GetLocation())))
+            If declaredBase IsNot Nothing AndAlso declaredBase.SpecialType = SpecialType.System_ValueType Then
+                Return
+            End If
 
+            ' Try to decrease noise by not complaining about the same type over and over again.
+            If Not _reportedErrorTypesMap.Add(namedTypeSymbol) Then
+                Return
+            End If
+
+            Dim location = If(syntaxNodeOpt Is Nothing, NoLocation.Singleton, syntaxNodeOpt.GetLocation())
+            If declaredBase IsNot Nothing Then
+                Dim useSiteDiagnostics As HashSet(Of DiagnosticInfo) = Nothing
+                declaredBase.AddUseSiteDiagnostics(useSiteDiagnostics)
+                If diagnostics.Add(location, useSiteDiagnostics) Then
+                    Return
                 End If
```

Comment:
Similar to C#, this code can be simplified to:
```
            Dim useSiteDiagnostic = declaredBase.GetUseSiteErrorInfo()
            If useSiteDiagnostic IsNot Nothing Then
                diagnostics.Add(useSiteDiagnostic, location)
                Return
            End If
``` 

There is also a helper in the binder ```Friend Shared Function ReportUseSiteError(diagBag As DiagnosticBag, syntax As SyntaxNodeOrToken, symbol As Symbol) As Boolean``` that can be cloned to work with location instead of SyntaxNodeOrToken.